gui/notes/index.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @module notes
+ * @flow
+ */
+
+const { models } = require('cozy-client')
+
+const path = require('path')
+const metadata = require('../../core/metadata')
+const {
+  CozyDocumentMissingError,
+  CozyMissingPermissionsError,
+  UnreachableError
+} = require('../../core/remote/errors')
+
+const logger = require('../../core/utils/logger')
+const log = logger({
+  component: 'Notes'
+})
+
+/*::
+import { Shell } from 'electron'
+import { App } from '../../core/app'
+import { Config } from '../../core/config'
+import { Pouch } from '../../core/pouch'
+import { Remote } from '../../core/remote'
+import type { Metadata } from '../../core/metadata'
+
+type CozyFile = Object
+*/
+
+/**
+ * computeNoteURL - Constructs a link to the Note within the web app
+ *
+ * @param {object} Options               Object of options
+ * @param {string}   options.noteId        Id of the CouchDB document representing the given note
+ * @param {string}   options.cozyUrl       Base URL of the cozy, eg. cozy.tools or test.mycozy.cloud
+ * @param {string}   options.subDomainType Whether the cozy is using flat or nested subdomains. Defaults to flat.
+ *
+ * @return {string} Generated URL
+ */
+const computeNoteURL = async ({ noteId, cozyUrl, subDomainType }) => {
+  const slug = 'notes'
+  const url = new URL(cozyUrl)
+  const nativePath = `/n/${noteId}`
+  url.host =
+    subDomainType === 'nested'
+      ? `${slug}.${url.host}`
+      : url.host
+          .split('.')
+          .map((x, i) => (i === 0 ? x + '-' + slug : x))
+          .join('.')
+  url.hash = nativePath
+  return url.toString()
+}
+
+const options = async (
```

Comment:
You don't use the new route from the stack that works also for notes from a shared directory (https://github.com/cozy/cozy-stack/pull/2345). Is it on purpose?

Reply:
I didn't know it was implemented already :). When do you think it's gonna be merged so I can test it?